src/main/java/seedu/address/model/roles/Role.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.roles;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Role in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidRoleName(String)}
+ */
+public class Role {
+
+    public static final String MESSAGE_CONSTRAINTS = "Role names should be alphanumeric with whitespace";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}\\s]+";
```

Comment:
Here's where I fixed "Fixed Roles not being able to have spaces (contradicts examples in the UG)"